Mapping/Loader/YamlLoader.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace Algolia\AlgoliaSearchBundle\Mapping\Loader;
+
+use Algolia\AlgoliaSearchBundle\Mapping\Description;
+use Algolia\AlgoliaSearchBundle\Mapping\Index;
+use Algolia\AlgoliaSearchBundle\Mapping\IndexIf;
+use Algolia\AlgoliaSearchBundle\Mapping\Method;
+use Algolia\AlgoliaSearchBundle\Mapping\Property;
+use Symfony\Component\Yaml\Yaml;
+
+class YamlLoader extends AbstractFileLoader
+{
+    protected function loadMetadataFromFile(\ReflectionClass $class, $file)
+    {
+        $config = Yaml::parse(file_get_contents($file));
+
+        if (!isset($config[$name = $class->name])) {
+            throw new \RuntimeException(sprintf('Expected metadata for class %s to be defined in %s.', $class->name, $file));
+        }
+
+        $config = $config[$name];
+        $className = $this->removeProxy($name);
+        $description = new Description($class);
+
+        $this->extractClassMetadatas($config, $className, $description);
+
+        $this->extractPropertyMetadata($class, $className, $config, $description);
+
+        $this->extractMethodMetadata($class, $className, $config, $description);
+
+        return $description;
+    }
+
+    /**
+     * @param $config
+     * @param $className
+     * @param $description
+     */
+    protected function extractClassMetadatas($config, $className, Description $description)
+    {
+        $classMetadata = [];
+
+        $classMetadata['perEnvironment'] = true;
+        if (isset($config['perEnvironment'])) {
+            $classMetadata['perEnvironment'] = $config['perEnvironment'];
+        }
+
+        $classMetadata['autoIndex'] = true;
+        if (isset($config['autoIndex'])) {
+            $classMetadata['autoIndex'] = $config['autoIndex'];
+        }
+
+        if (isset($config['algoliaName'])) {
+            $classMetadata['algoliaName'] = $config['algoliaName'];
+        }
+
+        foreach (Index::$algoliaSettingsProps as $field) {
+            if (isset($config[$field])) {
+                $classMetadata[$field] = $config[$field];
+            }
+        }
+
+        $index = new Index();
+        $index->setAlgoliaNameFromClass($className);
+        $index->updateSettingsFromArray($classMetadata);
+        $description->setIndex($index);
+    }
+
+    /**
+     * @param \ReflectionClass $class
+     * @param $className
+     * @param $config
+     * @param $description
+     */
+    protected function extractPropertyMetadata(\ReflectionClass $class, $className, $config, Description $description)
+    {
+        foreach ($class->getProperties() as $property) {
+            if ($className !== $property->class) {
```

Comment:
What is the intention of skipping inherited properties? This requires to redefine all properties to be indexed in an extended class. This is especially tricky if a project depends f.e. on entities from a third party project. At the moment I can see only a lot of not required boilerplate code coming up due to this restriction...?!